sos/plugins/openstack_cinder.py

```diff
@@ -27,8 +27,10 @@ class OpenStackCinder(Plugin):
     plugin_name = "openstack_cinder"
     profiles = ('openstack', 'openstack_controller')
 
-    option_list = [("db", "gathers openstack cinder db version", "slow",
-                    False)]
+    option_list = [
+        ("db", "gathers openstack cinder db version", "slow", False),
+        ("exclude", "Ignore files matching this path spec", "", "")
+    ]
```

Comment:
> So HPE customers will end up with glance-api-json.log for example, but I don't know what the 
> equivalent Red Hat ones would be called

I have a feeling I've seen names like `glance-api-json.log` in reports in the past, so hopefully there's some common ground.

It sounds like there may be two things needed here - a way to separate out (and drop) the JSON logs, and support for handling variations in the name, or name pattern, across distros or deployments.

If the names commonly vary from deployment to deployment then we normally handle that via some kind of detection process: ideally calling a program, but resorting to parsing configuration files or directory listings as the last resort (the `sar` plugin has some pretty hideous logic for interpreting /var/log/sa for e.g.).

There are a few plugins that enumerate objects and then dump information on them - historically disks and network interfaces for e.g., but quite a few of the virt and container plugins behave this way. If there are commands that can output the file names or paths then it's straightforward to add them.

Otherwise if the names are set by the distribution and not usually changed then they are controlled by the sos policy: this probes the host at start up and guesses what distribution is running. I'm not familiar with the HPE OpenStack distribution but if it includes the host Linux build (i.e. it doesn't run on top of RHEL, or Ubuntu or another distro), then it's probably worth adding a policy for it - this will allow separate branding strings, package manager, path, and file system configurations to be applied, and different defaults to be enabled. Plugins can also react to this and use policy information to construct or obtain paths and other strings.

Another pull request came in today for a plugin to collect Java JAR information in JSON format; it's a bit tangential to this specific problem, but as they're both dealing with adding JSON IO to plugins I'd like to see if we can come up with an interface that can cover both uses.